hooks/useAsyncLoad.ts

```diff
@@ -0,0 +1,28 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+
+export default function useAsyncLoad<T>(fn: () => Promise<T>) {
+  const funcRef = useRef<typeof fn>(fn);
+  const [data, setData] = useState<T>();
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    funcRef.current = fn;
+  }, [fn]);
+
+  const refetch = useCallback(async () => {
+    setIsLoading(true);
+    try {
+      setData(await funcRef.current());
+    } catch {
+      setData(undefined);
```

Comment:
maybe add a `const [error, setError] = useState<Error | null>(null)` and handle the error here